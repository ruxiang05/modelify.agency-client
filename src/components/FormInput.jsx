import React from 'react';
import PropTypes from 'prop-types';

const FormInput = ({
  title, name, type, value, handleChange, placeholder,
}) => (
  <div className="form-input">
    <label htmlFor={name}>{title}</label>
    <input
      id={name}
      name={name}
      type={type}
      value={value}
      onChange={handleChange}
      placeholder={placeholder}
    />
  </div>
);

FormInput.defaultProps = {
  value: '',
  placeholder: '',
};

FormInput.propTypes = {
  value: PropTypes.string,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};
export default FormInput;
