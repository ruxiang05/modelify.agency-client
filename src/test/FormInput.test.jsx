import React from 'react';
import { shallow } from 'enzyme';
import FormInput from '../components/FormInput';

const input = {
  title: 'title',
  name: 'name',
  type: 'text',
  value: '',
  handleChange: jest.fn(),
  placeholder: 'placeholder',
};

describe('FormInput', () => {
  it('should render FormInput', () => {
    const wrapper = shallow(<FormInput {...input} />);
    expect(wrapper).toMatchSnapshot();
  });
});
