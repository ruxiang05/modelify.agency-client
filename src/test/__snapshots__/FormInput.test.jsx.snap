// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render FormInput 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput
    handleChange={[MockFunction]}
    name="name"
    placeholder="placeholder"
    title="title"
    type="text"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="name"
        >
          title
        </label>,
        <input
          id="name"
          name="name"
          onChange={[MockFunction]}
          placeholder="placeholder"
          type="text"
          value=""
        />,
      ],
      "className": "form-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": "title",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "name",
          "name": "name",
          "onChange": [MockFunction],
          "placeholder": "placeholder",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="name"
          >
            title
          </label>,
          <input
            id="name"
            name="name"
            onChange={[MockFunction]}
            placeholder="placeholder"
            type="text"
            value=""
          />,
        ],
        "className": "form-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": "title",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "name",
            "name": "name",
            "onChange": [MockFunction],
            "placeholder": "placeholder",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
